config/routes.rb

```diff
@@ -44,4 +46,13 @@
   get 'profile', to: 'profile#show'
   post 'profile/switch-client', to: 'profile#switch_client'
   post 'profile/update-role', to: 'profile#update_role'
+
+  get '/component/:component_type/:component_id/certificate/:id', to: 'user_journey#view_certificate', as: 'view_certificate'
```

Comment:
would be nice to name the params fully
```suggestion
  get '/component/:component_type/:component_id/certificate/:certificate_id', to: 'user_journey#view_certificate', as: 'view_certificate'
```